src/sparsezoo/v2/objects/model_objects.py

```diff
@@ -114,3 +119,73 @@ def _validate_model(self, model: onnx.ModelProto) -> bool:
                         f"Found keys in numpy {key_type}: {list(numpy_dict.keys())}."
                     )
         return True
+
+
+class SelectDirectory(Directory):
+    """
+    Object that represents a directory from which the
+    user can fetch the contained directories and file
+    by key values.
+
+    :param files: list of files contained within the SelectDirectory
+    :param name: name of the NumpyDirectory
+    :param path: path of the NumpyDirectory
+    :param url: url of the NumpyDirectory
+    :param owner_path: path of the parent SelectDirectory
+    """
+
+    def __init__(
+        self,
+        files: List[File],
+        name: str,
+        path: Optional[str] = None,
+        url: Optional[str] = None,
+        owner_path: Optional[str] = None,
+    ):
+        self._default, self._available = None, None
+
+        super().__init__(
+            files=files, name=name, path=path, url=url, owner_path=owner_path
+        )
+
+        self.files_dict = self.files_to_dictionary()
+
+    def __getitem__(self, key):
+        file = self.files_dict[key]
+        file.get_path()
+        return file
```

Comment:
So the only thing this SelectDirectory has over Directory is this `__getitem__`? Could this be added to Directory instead of adding a new class?

Reply:
Well, not only. `SelectDirectory` basically extends the `Directory` with the possibility to 
- select default subdirectories through the `default` property or "dict-like" key selection
- looking up available subdirectories through `available` property.

I feel that those two classes are distinct enough so that their separate existence is justified. `SelectDirectory` basically is a special, advanced case of `Directory`. I'm confident that not all directories should be `SelectDirectories`.

Reply:
Ahh yeah, esp the default property since that specializes the directory structure to contain very specific things.

Thanks for the clarifications!